cmd/promtool/unittest.go

```diff
@@ -91,7 +104,7 @@ func ruleUnitTest(filename string, queryOpts promqltest.LazyLoaderOpts, run *reg
 	}
 
 	evalInterval := time.Duration(unitTestInp.EvaluationInterval)
-
+	ts.Settime(time.Now().Format("2006-01-02T15:04:05"))
 	// Giving number for groups mentioned in the file for ordering.
 	// Lower number group should be evaluated before higher number group.
 	groupOrderMap := make(map[string]int)
```

Comment:
Group order error is not handled on line 113 `return []error{fmt.Errorf("group name repeated in evaluation order: %s", gn)}` .

See for yourself, modify cmd/promtool/testdata/at-modifier-test.yml to have:
```
rule_files:
  - at-modifier.yml

tests:
  - input_series:
      - series: "requests{}"
        values: 1

group_eval_order:
  - at-modifier
  - at-modifier
```

This has the output:
```
$ ./promtool test rules --junit=out.xml cmd/promtool/testdata/at-modifier-test.yml 
  FAILED:
group name repeated in evaluation order: at-modifier


$ cat out.xml 
<testsuites><testsuite name="cmd/promtool/testdata/at-modifier-test.yml" tests="0" failures="0" errors="0" skipped="0" timestamp="2024-07-29T09:01:12"></testsuite></testsuites>
```

Which has no error. So I think we should have something like:
```
			err := fmt.Errorf("group name repeated in evaluation order: %s", gn)
			ts.Abort(err)
			return []error{err}
```

Reply:
Good spot. At some point (a future PR), it would be nice to avoid using the `error` type as the return value, it should be returning something wrapping error, then `ruleUnitTest` wouldn't need to have knowledge of junit. (I have reasons to want to do this to clean up the promtool text output)

To be clear I'm just mentioning as a possibility here, this approach is fine for this PR.

Reply:
yeah we need to think about this thing !